### ISSUE TITLE
any way to implement "rodeo" mechanic here?

### ISSUE THREAD

**Issue by ghost** [EnderBoy9217/Grunt-Mode#14]
i noticed that as a grunt u cannot rodeo the titans.. is there a way to set the mod so we can use that mechanic? i tried looking into the files but no luck. and also if it's possible climbing over the ally robots.. although, i havent tried that yet, so maybe that one is possible and didint know.

**Comment by dbyale** [comment]
Yes, this is intentional as rodeoing Titans as certain grunt classes, especially spectre classes, will crash the game for yourself and everyone else connected to the server. While it might be possible to partially re-implement, it wasn't something I thought important enough to spend extra time on. I also thought it made sense that grunts couldn't rodeo as they don't have jump-kits, even though as a side effect pilot classes can't rodeo either.

I believe the rodeo mechanic is disabled by making the titans unrodeoable entirely (if any rodeo animation starts with an invalid model, the game crashes, so kicking people off rodeo won't work fast enough).

If you removed that extra disabled you should be able to rodeo, however specific classes will crash when rodeoing.

I'm happy to help you modify it any way you'd like, but it is a codebase I primarily worked on almost 2 years ago, so it's a little messy but it works.

**Comment by dbyale** [comment]
Also, climbing allied robots and hacking spectres both count as rodeoing as well, and since they use similar animations they face the same crashing issue.

**Comment by ghost** [comment]
mm yikes.. ah well.. thx for the info.. right now, ur mod is the only full pve option i have.. after northstar updated the client with the "switching teams" feature.. mods like "attrition extended" stopped working..
